### PR TITLE
[CAS-1421] Docs for avatar customization

### PR DESCRIPTION
**Pull request by leandroBorgesFerreira**
### 🎯 Goal

Document how to configure avatar

### 🧪 Testing

run: `npx stream-chat-docusaurus -i -s`. In the root folder and read the documentation

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-team or #compose-chat-sdk-team) (required)
- [x] PR targets the `develop` branch

#### Code & documentation
- ~[ ] Changelog is updated with client-facing changes~
- ~[ ] New code is covered by unit tests~
- ~[ ] Comparison screenshots added for visual changes~
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF
giphy

### CODE REVIEW

File: docusaurus/docs/Android/03-ui/03-chatui.mdx [hunk 1]
```diff
@@ -75,16 +75,45 @@ ChatUI.mimeTypeIconProvider = MimeTypeIconProvider { mimeType ->
...
+public open suspend fun createUserBitmap(
+        user: User,
+        style: AvatarStyle,
+        @Px avatarSize: Int,
+    ): Bitmap? {
+        return createDefaultUserBitmap(user, style, avatarSize)
+    }
+```
+
+**If you only would like  to change the gradient colors for the avatar**, you can use `stream_ui_avatar_gradient_colors`. You can override this color in your color xml file like:
```
Comment (skydoves) [inline docusaurus/docs/Android/03-ui/03-chatui.mdx:109]:
```suggestion
**If you only would like to change the gradient colors for the avatar**, you can use `stream_ui_avatar_gradient_colors`. You can override this color in your color xml file like: 
```

**Review by skydoves — APPROVED** [review]
Looks good to me. 👍

**Review by samiuelson — APPROVED** [review]

File: docusaurus/docs/Android/03-ui/03-chatui.mdx [hunk 2]
```diff
@@ -75,16 +75,45 @@ ChatUI.mimeTypeIconProvider = MimeTypeIconProvider { mimeType ->
 }
 ```
 
-### Customizing Avatar Bitmap Loading
+### Customizing Avatar
 
-Overriding the `AvatarBitmapFactory` allows you to add custom loading logic for the channel and user avatars displayed in `AvatarView`. 
+#### Customizing Avatar Using Styles
+You can configure avatar shape, border width, online indicator and other aspects using [AvatarStyle](https://github.com/GetStream/stream-chat-android/blob/main/stream-chat-android/src/main/java/com/getstream/sdk/chat/view/messages/AvatarStyle.kt). Changing the shape and border radius, it is possible to create this kind of avatar:
```
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/03-chatui.mdx:81]:
```suggestion
You can configure avatar shape, border width, online indicator and other aspects using [AvatarStyle](https://github.com/GetStream/stream-chat-android/blob/main/stream-chat-android/src/main/java/com/getstream/sdk/chat/view/messages/AvatarStyle.kt). You can create this kind of avatar by changing the shape and border-radius: 
```

File: docusaurus/docs/Android/03-ui/03-chatui.mdx [hunk 3]
```diff
@@ -75,16 +75,45 @@ ChatUI.mimeTypeIconProvider = MimeTypeIconProvider { mimeType ->
...
-### Customizing Avatar Bitmap Loading
+### Customizing Avatar
 
-Overriding the `AvatarBitmapFactory` allows you to add custom loading logic for the channel and user avatars displayed in `AvatarView`. 
+#### Customizing Avatar Using Styles
+You can configure avatar shape, border width, online indicator and other aspects using [AvatarStyle](https://github.com/GetStream/stream-chat-android/blob/main/stream-chat-android/src/main/java/com/getstream/sdk/chat/view/messages/AvatarStyle.kt). Changing the shape and border radius, it is possible to create this kind of avatar: 
+
+![Light_mode](../assets/square_avatar.png)
+
+It is also possible to change the background and the picture using `AvatarBitmapFactory`.
```
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/03-chatui.mdx:85]:
```suggestion
You can also change the background and the picture using `AvatarBitmapFactory`.
```

File: docusaurus/docs/Android/03-ui/03-chatui.mdx [hunk 4]
```diff
@@ -75,16 +75,45 @@ ChatUI.mimeTypeIconProvider = MimeTypeIconProvider { mimeType ->
...
+
+Overriding the `AvatarBitmapFactory` allows you to add custom logic for the channel's and user's avatars displayed using `AvatarView`. 
 
 The factory has a pair of methods for both user and channel avatars, such as `createUserBitmap` and `createDefaultUserBitmap`. The first should load the avatar from the network, this is attempted first when an avatar is requested. If the first method fails, the `Default` variant is called as a fallback. This should have an implementation that never fails, and always returns a valid bitmap.
 
 :::note
 The methods mentioned above are suspending functions. If you don't use coroutines, you'll find variants with `Blocking` in their name that you can use instead.
 :::
 
+**If you don't want to use avatars from internet and only use the default avatar**, you can call `createDefaultUserBitmap` inside `createUserBitmap` in your custom implementation of `AvatarBitmapFactory`
```
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/03-chatui.mdx:97]:
```suggestion
**If you don't want to use avatars from the internet and only use the default avatar**, you can call `createDefaultUserBitmap` inside `createUserBitmap` in your custom implementation of `AvatarBitmapFactory`
```

File: docusaurus/docs/Android/03-ui/03-chatui.mdx (hunk 1 above)
Comment (adasiewiczr) [reply docusaurus/docs/Android/03-ui/03-chatui.mdx:109]:
```suggestion
**If you only would like to change the gradient colors for the default avatar**, you can use `stream_ui_avatar_gradient_colors`. You can override this color in your color xml file like: 
```

**Review by adasiewiczr — CHANGES_REQUESTED** [review]

File: docusaurus/docs/Android/03-ui/03-chatui.mdx [hunk 5]
```diff
@@ -75,16 +75,68 @@ ChatUI.mimeTypeIconProvider = MimeTypeIconProvider { mimeType ->
 }
 ```
 
-### Customizing Avatar Bitmap Loading
+### Customizing Avatar
+The SDK uses `AvatarView` in the lists of channels, users and in the message lists. The avatar is a small picture that identifies an specific channel or user.
```
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/03-chatui.mdx:79]:
```suggestion
The `AvatarView` is used in the lists of channels, users, and messages. The avatar is a small picture that identifies a specific channel or user. 
```

File: docusaurus/docs/Android/03-ui/03-chatui.mdx [hunk 6]
```diff
@@ -75,16 +75,68 @@ ChatUI.mimeTypeIconProvider = MimeTypeIconProvider { mimeType ->
 }
 ```
 
-### Customizing Avatar Bitmap Loading
+### Customizing Avatar
+The SDK uses `AvatarView` in the lists of channels, users and in the message lists. The avatar is a small picture that identifies an specific channel or user. 
 
-Overriding the `AvatarBitmapFactory` allows you to add custom loading logic for the channel and user avatars displayed in `AvatarView`. 
+For `Channel`, it is used the property Channel.image and for User, it is used `User.image`. The image in the AvatarView, will be the image's url in those fields:
```
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/03-chatui.mdx:81]:
```suggestion
The image in the `AvatarView` is being displayed based on `image` property, present in both `Channel` and `User` objects:
```

File: docusaurus/docs/Android/03-ui/03-chatui.mdx [hunk 7]
```diff
@@ -75,16 +75,68 @@ ChatUI.mimeTypeIconProvider = MimeTypeIconProvider { mimeType ->
...
-### Customizing Avatar Bitmap Loading
+### Customizing Avatar
+The SDK uses `AvatarView` in the lists of channels, users and in the message lists. The avatar is a small picture that identifies an specific channel or user. 
 
-Overriding the `AvatarBitmapFactory` allows you to add custom loading logic for the channel and user avatars displayed in `AvatarView`. 
+For `Channel`, it is used the property Channel.image and for User, it is used `User.image`. The image in the AvatarView, will be the image's url in those fields: 
+
+![Default avatar](../assets/default_channel.png)
+
+If there's no url to be displayed, the SDK falls back to a gradient color with initials for the channel or user:
```
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/03-chatui.mdx:85]:
```suggestion
The `AvatarView` will use default gradient color with initials if the `image` property cannot be loaded:
```

File: docusaurus/docs/Android/03-ui/03-chatui.mdx [hunk 8]
```diff
@@ -75,16 +75,68 @@ ChatUI.mimeTypeIconProvider = MimeTypeIconProvider { mimeType ->
...
+
+Overriding the `AvatarBitmapFactory` allows you to add custom logic for the channel's and user's avatars displayed using `AvatarView`. 
 
 The factory has a pair of methods for both user and channel avatars, such as `createUserBitmap` and `createDefaultUserBitmap`. The first should load the avatar from the network, this is attempted first when an avatar is requested. If the first method fails, the `Default` variant is called as a fallback. This should have an implementation that never fails, and always returns a valid bitmap.
 
 :::note
 The methods mentioned above are suspending functions. If you don't use coroutines, you'll find variants with `Blocking` in their name that you can use instead.
 :::
 
+**If you don't want to use avatars from the internet and only use the default avatar**, you can call `createDefaultUserBitmap` inside `createUserBitmap` in your custom implementation of `AvatarBitmapFactory`
```
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/03-chatui.mdx:107]:
```suggestion
**If you don't want to use avatars from the internet and only use the default avatar**, you can call `createDefaultUserBitmap` inside `createUserBitmap` in your custom implementation of `AvatarBitmapFactory`:
```

File: docusaurus/docs/Android/03-ui/03-chatui.mdx [hunk 9]
```diff
@@ -75,16 +75,68 @@ ChatUI.mimeTypeIconProvider = MimeTypeIconProvider { mimeType ->
...
+public open suspend fun createUserBitmap(
+        user: User,
+        style: AvatarStyle,
+        @Px avatarSize: Int,
+    ): Bitmap? {
+        return createDefaultUserBitmap(user, style, avatarSize)
+    }
+```
+
+This way use default avatar with gradient colors and initials for the users.
```
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/03-chatui.mdx:119]: I think we don't need it. You already described it above the snippet
```suggestion
```

File: docusaurus/docs/Android/03-ui/03-chatui.mdx [hunk 10]
```diff
@@ -75,16 +75,68 @@ ChatUI.mimeTypeIconProvider = MimeTypeIconProvider { mimeType ->
...
+
+**If you only would like to change the gradient colors for the default avatar**, you can use `stream_ui_avatar_gradient_colors`.
+
+The normal color set include a variaty of colors:
```
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/03-chatui.mdx:123]:
```suggestion
The default color set includes a variety of colors: 
```

File: docusaurus/docs/Android/03-ui/03-chatui.mdx [hunk 11]
```diff
@@ -75,16 +75,68 @@ ChatUI.mimeTypeIconProvider = MimeTypeIconProvider { mimeType ->
...
+
+This way use default avatar with gradient colors and initials for the users.
+
+**If you only would like to change the gradient colors for the default avatar**, you can use `stream_ui_avatar_gradient_colors`.
+
+The normal color set include a variaty of colors: 
+
+![Colorful avatars](../assets/colorful_avatars.png)
+
+You can override this color in your color xml file. And expand of reduce the number of supported color, like the example
```
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/03-chatui.mdx:127]:
```suggestion
The set can be overridden in the `color.xml` file - you can expand or reduce the number of supported colors, like in the example below:
```

**Review by adasiewiczr — COMMENTED** [review]
Looks great overall. Added small suggestions

**Review by adasiewiczr — APPROVED** [review]

**Pull request merged**